infra/Pos/Network/Windows/DnsDomains.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+#if defined(mingw32_HOST_OS)
+#define WINDOWS
+#endif
+
+module Pos.Network.Windows.DnsDomains (
+#ifdef WINDOWS
+    getWindowsDefaultDnsServer
+#endif
+    ) where
+
+#ifdef WINDOWS
+import           Universum
+
+import           Foreign.C            (CString, peekCString)
+
+foreign import ccall "getWindowsDefDnsServer" getWindowsDefDnsServer :: IO CString
+
+getWindowsDefaultDnsServer :: IO String
+getWindowsDefaultDnsServer = getWindowsDefDnsServer >>= peekCString
```

Comment:
Were it not a `static`ally allocated string, desugaring the bind into a `do` and adding a bang on the string would have prevented it, correct?